tracer/test/Datadog.Trace.Tests/Agent/MessagePack/SpanMetaStructTests.cs

```diff
@@ -103,7 +103,7 @@ public class SpanMetaStructTests
             }
        };
 
-    [MemberData(nameof(Data))]
+    [MemberData(nameof(Data), DisableDiscoveryEnumeration = true)]
```

Comment:
Don't we want to fix the underlying issue instead of disabling the warning?

See https://github.com/DataDog/dd-trace-dotnet/pull/5089

Reply:
I mean, maybe? But I they are useless warnings.

> which makes me thing it will only run on tests case instead of all of them.

I don't think so, the tests all run locally on the CLI and via the VS Test Explorer.

What is happening is that xunit (I think or maybe the test host) is attempting to enumerate all of the individual theory data lines just so that you could run an individual Theory data by itself instead of the entire test.

That is what the `Non-serializable data ('System.Object[]') found for 'Datadog.Trace.Tests.Tagging.ActivityTagsTests.Tags_ShouldBe_PlacedInMetricsOrMeta'; falling back to single test case.` is ultimately meaning.

Reply:
I didn't see that we have been fixing them to be serializable previously

But I'll swap to fix it to be consistent.

I am curious if that enumeration is slowing our tests down 🤔

Reply:
You're right it still runs all the cases at run time, so _meh_. Thanks for looking into this!

Reply:
Yeah, the analysis is 100 correct, it just gives a worse local dev experience, I only learned that after raising an issue for it here 😄  https://github.com/xunit/xunit/issues/2866

> I am curious if that enumeration is slowing our tests down

My strong suspicion is that it's a drop in the ocean and not worth worrying about 😃